pkg/api/norman/customization/aks/listers.go

```diff
@@ -326,3 +415,13 @@ func encodeOutput(result interface{}) ([]byte, int, error) {
 
 	return data, http.StatusOK, err
 }
+
+func buildUpgradeableVersionsMap(upgradeProfile containerservice.OrchestratorVersionProfile) map[string]bool {
+	rval := make(map[string]bool, 0)
+
+	for _, profile := range *upgradeProfile.Upgrades {
```

Comment:
If this profile has all the info we need why are we doing all the other processing?

Reply:
This profile will not contain future versions that are not supported yet.